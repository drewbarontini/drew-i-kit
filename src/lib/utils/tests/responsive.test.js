import { responsive } from '..';

describe('responsive()', () => {
  it('should return the correct breakpoints', () => {
    const props = {
      breakpoints: [
        { size: 's', styles: () => `color: red;` },
        { size: 'l', styles: () => `color: green;` },
        { size: '600px', styles: () => `color: blue;` },
      ],
    };
    const query = responsive(props).join('');
    expect(query).toContain('@media');
    expect(query).toContain('max-width');
    expect(query).toContain('600px');
    expect(query).toContain('color: red;');
    expect(query).toContain('color: green;');
    expect(query).toContain('color: blue;');
  });
});
