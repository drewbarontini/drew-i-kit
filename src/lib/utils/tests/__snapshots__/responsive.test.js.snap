// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hideAndShow() should generate a hiding and showing query 1`] = `
Array [
  Object {
    "size": "l",
    "styles": [Function],
  },
  Object {
    "size": "s",
    "styles": [Function],
  },
]
`;

exports[`hideAndShow() should generate a hiding query 1`] = `
Array [
  Object {
    "size": "s",
    "styles": [Function],
  },
]
`;
